vsc_pdf_template_transformer/lib/src/vsc_pdf_template_transformer_base.dart

```diff
@@ -1,6 +1,49 @@
-// TODO: Put public facing types in this file.
+import 'dart:convert';
+import 'package:pdf/pdf.dart';
+import 'package:pdf/widgets.dart' as pw;
+import 'package:vsc_pdf_template_transformer/models/text_pdf.dart';
 
-/// Checks if you are awesome. Spoiler: you are.
-class Awesome {
-  bool get isAwesome => true;
+import '../utils/node.dart';
+import '../utils/widget_builder.dart';
+
+class Transformer {
+  get isReady => true;
+
+  static pw.Document buildPdf(
+      Node<String> treeRoot, Map<String, dynamic> data) {
+    final pdf = pw.Document();
+    pdf.addPage(pw.Page(
+        pageFormat: PdfPageFormat.a4,
+        build: (pw.Context context) {
+          var valueMap = json.decode(treeRoot.value);
+          var className = valueMap['className'];
+          print('---- root element is of a type ${className} ---');
+          WidgetBuilder? proxy =
+              getWidgetBuilder(valueMap); //TextPdf.fromJson(valueMap);
+          return pw.Center(
+            child: proxy != null
+                ? proxy!.buildWidget()
+                : pw.Text('Unsupported Component: ${className}'),
+          ); // Center
+        }));
+    return pdf;
+  }
+
+  //TODO: Extend with more types. Mirrors package not available in Flutter project so we can't create an instance of a class by name via reflection
+  static WidgetBuilder? getWidgetBuilder(Map<String, dynamic> valueMap) {
+    WidgetBuilder? result;
+    switch (valueMap['className']) {
```

Comment:
I like the idea of `getWidgetBuilder`, but instead of a switch which will get pretty lengthy, I suggest using a map, for example:

```dart
const widgetClassFromJson = {
  'TextPdf': TextPdf.fromJson,
// ...
```

Then this method just looks up the `fromJson` method in the map: 
```dart
final fromJson = widgetClassFromJson[className];
if (fromJson == null) // throw exception
result = fromJson(valueMap);
```

Reply:
Cool, works! Thank you!